### PR TITLE
Handle nullable differently to optional

### PR DESCRIPTION
**Pull request by ianagbip1oti**
This handles the reading of nullable fields.

Writing to come in future PR.

### CODE REVIEW

**Comment by CoreyShupe** [comment]
Are we going to do anything about filtering with jackson in this PR, seems appropriate.

**Comment by ianagbip1oti** [comment]
I thought about that, but there's no classes here where we actually write anything out. Instead I've included it with other code that adds a Data.toJsonString method and uses that to test the filter.

**Comment by CoreyShupe** [comment]
Does the Nullable and Optional correctly differentiate with Jackson at this point? Or is the filtering of the 2 in a diff piece.

**Comment by ianagbip1oti** [comment]
The filtering is elsewhere. This is why the description calls out that this handles the reading of nullable/optional. Filtering is only needed for writing.

**Comment by CoreyShupe** [comment]
Alright, looks good then, I'll approve.

File: src/codegen/ruby/def/field.rb [hunk 1]
```diff
@@ -41,6 +42,7 @@ def write(f)
     f.puts "  /**"
     f.puts "    * #{description}"
     f.puts "    */"
+    f.puts "  @javax.annotation.Nullable" if nullable?
```
Comment (CoreyShupe) [inline src/codegen/ruby/def/field.rb]: Do we really want to use javax.annotation or should we use the reactivex annotation?

Comment (ianagbip1oti) [reply src/codegen/ruby/def/field.rb]: what you said is about it. Immutables depends on jsr305 to make use of the annotations provided there (which is where javax.annotation.Nullable comes from)

Comment (CoreyShupe) [reply src/codegen/ruby/def/field.rb]: Alrighty, don't see anything else I have concerns about.

**Pull request merged**